packages/contracts-bedrock/src/L1/OptimismPortal2.sol

```diff
@@ -354,18 +358,46 @@ contract OptimismPortal2 is Initializable, ResourceMetering, ReinitializableBase
         // Intentionally empty.
     }
 
-    /// @notice Updates the ETHLockbox contract.
-    /// @dev    This function MUST be called atomically with `ETHLockbox.migrateLiquidity()`
-    ///         in the same transaction batch, or otherwise the OptimismPortal may not be able to
-    ///         unlock ETH from the ETHLockbox on finalized withdrawals.
+    /// @notice Allows the owner of the ProxyAdmin to migrate the OptimismPortal to use a new
+    ///         lockbox, point at a new AnchorStateRegistry, and start to use the Super Roots proof
+    ///         method. Primarily used for OptimismPortal instances to join the interop set, but
+    ///         can also be used to swap the proof method from Output Roots to Super Roots if the
+    ///         provided lockbox is the same as the current one.
+    /// @dev    It is possible to change lockboxes without migrating liquidity. This can cause one
+    ///         of the OptimismPortal instances connected to the new lockbox to not be able to
+    ///         unlock sufficient ETH to finalize withdrawals which would trigger reverts. To avoid
+    ///         this issue, guarantee that this function is called atomically alongside the
+    ///         ETHLockbox.migrateLiquidity() function within the same transaction.
```

Comment:
I recall discussion around this, but why do we not enforce it just by calling `_migrateLiquidity()` at the end?

Reply:
I was confusing `OptimismPortal.migrateLiquidity()` with `ETHLockbox.migrateLiquidity()`.